Test/Core/Components/CollisionEventReceiver.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+
+namespace Duality.Tests.Components
+{
+	/// <summary>
+	/// A collision listener that stores collision events in lists
+	/// </summary>
+	public class CollisionEventReceiver : Component, ICmpCollisionListener
+	{
+		public struct CollisionEvent
+		{
+			public readonly Component Sender;
+			public readonly CollisionEventArgs Args;
+
+			public CollisionEvent(Component sender, CollisionEventArgs args)
+			{
+				this.Sender = sender;
+				this.Args = args;
+			}
+		}
+
+		public readonly List<CollisionEvent> CollisionBegin = new List<CollisionEvent>();
```

Comment:
With three separate lists instead of a struct with a local "event type" enum, the order in which different events arrive is lost. Not a blocker for a merge (read as FYI, you decide), but probably not future-proof either.

Reply:
Ye the order of the events could be something that's important, good point. Actually when I think about it might as well check that in the ball test too? Pretty sure begin collision has to come before the others.

Reply:
> Actually when I think about it might as well check that in the ball test too? Pretty sure begin collision has to come before the others.

Yep, makes sense. If you check for this as well, the intended order would be `Begin`, `Solve` times N (any N is fine), `End`, and that order might show up multiple times in case the ball bounces off. For the simple test, maybe just set the `Restitution` values of both ball and platform to zero, so there should be no bouncing.

Reply:
Restitution is 0 by default but I will set it explicitly nonetheless in case that changes.

Reply:
Added a DebuggerDisplay attribute to CollisionEvent as well so you can see at once the type and the gameobject it collided with. If you prefer a ToString override to achieve this I can change that.